tailnet/controllers_test.go

```diff
@@ -1253,15 +1258,13 @@ func (f fakeCoordinatorClient) Close() error {
 	errs := make(chan error)
 	select {
 	case <-f.ctx.Done():
-		f.t.Error("timed out waiting to send close call")
-		return f.ctx.Err()
+		return timeoutOnFakeErr
```

Comment:
note: changes like this are because calling `t.Error()` after the test case finishes (and context expires in `t.Cleanup`) will panic and create a bunch of extra tests to fail.  So, we don't fail the test in these handlers, and instead pass a distinctive error.

Reply:
Nice! I assume it's a similar fix for https://github.com/coder/internal/issues/217 too

Reply:
Yeah, plus there was a real product bug in that one.